blog/sudoko_blog_structure.md

```diff
@@ -132,8 +132,37 @@ Explanation of kernels and incorporating rules.
 ## Loss Function
 
 <div align="center">
-    <img src="./images/colouredSudoku.png" width=200px />
-    <img src="./images/lossVisualIdea.png" width=400px />
+    <img src="./images/recolouredSudoku.png" width=200px/>
+</div>
+
+Given that the loss function is what drives the learning process of a neural network, we put a lot of time and thought into which one to use for this particular problem.
+
+The base loss function that we decided on in the end was CrossEntropy. This loss function is often used in multi-classification problems and sudoku can be thought of along these lines. For each blank cell, our model must look at the rest of the cells in the puzzle and determine what to fill it with. There are nine options for a sudoku puzzle so in essence the model is attempting to classify each cell and we take its best guess (the value with the highest probability) and use that to fill in the blank.
+
+### Preprocessing
+
+We settled on a loss function but we could not call it a day there. In order for the model to be able to produce good results, we decided to look at a few different methods of preprocessing the output data before passing it to the loss function.
+
+Our first method (compare_all) was our most straightforward. We looked at each number in the predicted sudoku grid and determined whether or not it was correct (comparing to the known values). We did not care if any value was already in the grid - we treated every value the same and each was either right or wrong.
+
+(The following diagrams aim to visualise how we compare sudoku cell values during preprocessing. The key is: Green - Correct, Red - Wrong, Yellow - Prefilled Correct, Purple - Prefilled Wrong)
```

Comment:
I think you also need to explain what the black line is and what the significance of things being over/under the line.